pkg/roll/update_test.go

```diff
@@ -138,19 +158,389 @@ func SimulateRollingUpgradeDisable(t *testing.T, full, nonew bool) {
 		Assert(t).IsTrue(len(old)+len(new) >= minimum, fmt.Sprintf("went below %d minimum nodes (nodes %v)\n", minimum, nodes))
 		Assert(t).IsTrue(len(new) <= target, fmt.Sprintf("went above %d target nodes (nodes %v)\n", target, nodes))
 		if len(new) == target {
-			Assert(t).AreEqual(rollAlgorithm(len(old), len(new), target, minimum), 0, "update should be done")
+			nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target, target, minimum)
+			Assert(t).AreEqual(nextRemove, 0, "update should be done, should remove nothing")
+			Assert(t).AreEqual(nextAdd, 0, "update should be done, should add nothing")
 			t.Logf("Simulation complete\n\n")
 			break
 		}
 
 		// calculate the next update
-		nextUpdate := rollAlgorithm(len(old), len(new), target, minimum)
-		t.Logf("Scheduling %d new out of %v eligible\n", nextUpdate, eligible)
-		Assert(t).AreNotEqual(nextUpdate, 0, "got noop update, would never terminate")
+		nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target, target, minimum)
+		Assert(t).AreEqual(nextRemove, nextAdd, "got asymmetric update, not expected for this fuzz test")
+		t.Logf("Scheduling %d new out of %v eligible\n", nextAdd, eligible)
+		Assert(t).AreNotEqual(nextAdd, 0, "got noop update, would never terminate")
 		// choose nodes from the eligible list, randomly, and put the new pod
 		// on them
-		for _, index := range rand.Perm(len(eligible))[:nextUpdate] {
+		for _, index := range rand.Perm(len(eligible))[:nextAdd] {
 			nodes[eligible[index]] = 1
 		}
 	}
 }
+
+type fakeHealthChecker struct {
```

Comment:
added in 3c5192ce10c0ad0e93fed3b2a7515c48c0aa3a53 though I left a TODO that replication tests should probably put some stuff in there. maybe I'll just do that now instead of leaving the TODO